llvm/lib/Transforms/Utils/InjectTLIMappings.cpp

```diff
@@ -56,6 +69,8 @@ static void addVariantDeclaration(CallInst &CI, const ElementCount &VF,
   Function *VecFunc =
       Function::Create(VectorFTy, Function::ExternalLinkage, VFName, M);
   VecFunc->copyAttributesFrom(CI.getCalledFunction());
+  removeIllegalAttributes(VecFunc, VectorFTy);
```

Comment:
Is it safe to just always remove the attributes if doing so would lead to incorrect results? Do the vector result and arguments now need explicit sign or zero-extending?

Reply:
Sure, I understand. But then is it safe to even use a vector mapping in that case? I'm just asking if perhaps this is the wrong fix, and really we shouldn't be vectorising at all?